overlord/devicestate/handlers_gadget.go

```diff
@@ -285,8 +391,12 @@ func (m *DeviceManager) doUpdateGadgetCommandLine(t *state.Task, _ *tomb.Tomb) e
 	// snap carries an update to the gadget assets and a change in the
 	// kernel command line
 
-	// kernel command line was updated, request a reboot to make it effective
+	if isSysOption {
+		logger.Debugf("change comes from system option, we do not reboot")
```

Comment:
we probably still want to mark the task Done already?

Reply:
Done